### ISSUE TITLE
Updates to resources intermittently not triggering jobs with trigger: true

### ISSUE THREAD

**Issue by saedalavinia** [concourse/concourse#3682]
## Bug Report

Sometimes (particularly after a bad state due to either a failed job or issues in deployments such as workers restarting) the pipeline goes into a weird state with the following characteristics: 

- Jobs that are listening on resources with `trigger: true` no longer trigger when new versions are available. This is true for at least `semver`,  `git`, and `s3`. Looking at the UI, the resources are checked correctly, but the jobs simply doesn't run. When you manually trigger the job, it properly picks up the latest version, but it does not trigger automatically. 

- Some Jobs don't run anymore, even when you manually start them. Interestingly, they properly get all the resources, but they just sit there waiting 
![pipeline](https://user-images.githubusercontent.com/5727936/55516018-19a75780-563a-11e9-8650-1bb922fa493e.JPG)

This might be related to [#2569](https://github.com/concourse/concourse/issues/2569)

The only way to get the pipeline running again is to delete and redeploy the pipeline. 

The issue happens intermittently. We only noticed it after migrating to 5.0.1 with `global_resources` enabled. But here is a general trend: 

1. Have a pipeline: 

```yml 

jobs:
- name: deploy
  serial: true
  plan:
  - get: repo
     trigger: true
  - put: counter
    params:
      bump: major
- name: do-something
  plan:
  - get: counter
    trigger: ((auto_deploy))
  - get: repo
     trigger: true
  - task: something 
```
2. Let it run just fine. Then let something bad happen. This could be within pipeline logic ( a job that fails) or at the concourse deployment level (worker running on a spot instance to be recreated or simply a temporary network issue). After that, the jobs no longer auto-trigger on new resource checks. 
We are yet to even correlated part 2 (the occurrence of something bad). 

## Expected Results

The pipelines to behave consistently 

## Actual Results

Pipelines stop triggering when new resource versions are available. Some jobs no longer run. 

## Additional Context

None at this point, but I will be starting to dig a bit deeper. 

## Version Info

* Concourse version: 5.0.1 
* Deployment type (BOSH/Docker/binary): Docker containers running on Kubernetes, deployed using helm. 
* Infrastructure/IaaS: AWS
* Browser (if applicable): N/A
* Did this used to work?
Yes absolutely. Prior to 5.0.1 this definitely worked or at least we never encountered it. The closest bug report to the one we are experiencing is 
This might be related to [#2569](https://github.com/concourse/concourse/issues/2569) which is reporting this behavior since 3.x. But we never had any issue until 5.0.1

**Comment by clarafu** [comment]
This might related to #2438 where the scheduler is stuck on a hanging resource check.

**Comment by saedalavinia** [comment]
@clarafu It may or may not be. For us, this seems to be a 5.x thing. We never had any problem with this before. Now, this is happening to multiple of our pipelines everyday. 
We are working to get a simple reproducible case

**Comment by saedalavinia** [comment]
Some more update on this issue. 

Originally we were able to simply our reproducible case to just the following pipeline:

```sh 
resources:

- name: semver-trigger
  type: semver
  source:
    initial_version: 0.0.0
    driver: s3
    region: us-east-1 
    bucket: ((bucket))
    access_key_id: ((aws_access_key_id))
    secret_access_key: ((aws_secret_access_key))
    key: ((key))
  
- name: app-zip
  type: s3
  source:
    driver: s3
    region: us-east-1 
    bucket:  ((bucket))
    access_key_id: ((aws_access_key_id))
    secret_access_key: ((aws_secret_access_key))
    key: ((key))

jobs: 

- name: trigger-manual-deployment
  serial: true
  plan:
  - put: semver-trigger
    params:
      bump: patch


- name: deploy-app1
  serial_groups:
  - serial-group-1
  plan:
  - aggregate:
    - get: semver-trigger
      trigger: true
    - get: app-zip
  - task: try-things
    config:
      platform: linux
      image_resource:
        type: docker-image
        source:
          repository: alpine
          tag: latest
      run:
        path: sh
        args:
          - -uxec
          - |
            sleep 10
            echo "first job"
  serial: true

- name: deploy-app2
  serial: true
  serial_groups:
  - serial-group-1
  plan:
  - aggregate:
    - get: semver-trigger
      trigger: true
    - get: app-zip
  - task: try-things
    config:
      platform: linux
      image_resource:
        type: docker-image
        source:
          repository: alpine
          tag: latest
      run:
        path: sh
        args:
          - -uxec
          - |
            sleep 10
            echo "second job"


- name: deploy-app3
  serial: true
  plan:
    - aggregate:
        - get: semver-trigger
          trigger: true
        - get: app-zip
    - task: try-things
      config:
        platform: linux
        image_resource:
          type: docker-image
          source:
            repository: alpine
            tag: latest
        run:
          path: sh
          args:
            - -uxec
            - |
              sleep 10
              echo "third job"

```

However, after attempting to reproduce this issue on BUCC deployment, we were not able to reproduce it. That made us look at our deployment and we noticed that out of the two HA web pods, one of them was 7 days old, while the other one was just 2 days old.  After restarting the older instance, things started working again. 

That makes me think the issue is related to the HA web cluster

**Comment by MirahImage** [comment]
Just wanted to add that we (the PCF Redis team) are seeing this as well. We've noticed that the git resource has an updated version, but the job is not triggering on the new version.

**Comment by vito** [comment]
@MirahImage Is it fixed by a restart of the `web` nodes? Could you take a stack dump beforehand? See [Taking a Stack Dump](https://github.com/concourse/concourse/wiki/Taking-a-Stack-Dump). Thanks!

**Comment by MirahImage** [comment]
@vito We did restart the web node and that did fix it. Unfortunately, we didn't take a stack dump beforehand, but if it happens again, I will make sure we take a stack dump this time.

**Comment by MirahImage** [comment]
Hi @vito, we are having issues again and have taken a dump. What is the best way to get the dump to you?

**Comment by vito** [comment]
@MirahImage Does it fit in a gist? https://gist.github.com

**Comment by itsdalmo** [comment]
I believe I might be running into this bug as well. We have a `git-resource` for 4 subdirectories in the same repository, and one of them has stopped discovering new versions/triggering jobs. The resource is reporting that it's checking successfully, but the last check is 7h age even though it's set to check every minute. 

Manually triggering a check in the UI seems to be working (wheel spins and then turns into a green checkmark), but it does not discover any of the missing commits, and still reads `checked 7h 19m ago` (I even refreshed the page to make sure).

Running `fly -t <team> check-resource -r <pipeline>/<resource>` produces the following error (same result when specifying `-f "ref:<ref>"`:
```
error: Unexpected Response
Status: 504 Gateway Time-out
Body:
<html>
<head><title>504 Gateway Time-out</title></head>
<body bgcolor="white">
<center><h1>504 Gateway Time-out</h1></center>
</body>
</html>
```

However, `check-resource` works fine for the other `git` resources that use the same git configuration, and the only difference is that they track a different path.

I'm only logging on the `error` level for the ATC, and for the `<stuck-resource>` I can see the following error message from yesterday, and after that it is quiet (even with the manually triggered checks I attempted): 

```json
{
    "timestamp": "2019-05-15T17:16:48.760253924Z",
    "level": "error",
    "source": "atc",
    "message": "atc.pipelines.radar.scan-resource.interval-runner.tick.failed-to-check",
    "data": {
        "error": "Timed out after 1h0m0s while checking for new versions - perhaps increase your resource check timeout?",
        "pipeline": "<pipeline>",
        "resource": "<stuck-resource>",
        "session": "18.327.9.1.985",
        "team": "<team>"
    }
}
```

After the error message above, all I can see are the following two error messages (several hours later):

```json
{
    "timestamp": "2019-05-16T02:04:41.417389775Z",
    "level": "error",
    "source": "atc",
    "message": "atc.pipelines.scheduler.tick.scheduling.save-next-input-mapping.failed-to-save-independent-input-mapping",
    "data": {
        "error": "pq: insert or update on table \"independent_build_inputs\" violates foreign key constraint \"independent_build_inputs_resource_config_version_id_fkey\"",
        "pipeline": "<pipeline>",
        "session": "18.336.11947.3.6",
        "team": "<team>"
    }
}
```

```json
{
    "timestamp": "2019-05-16T08:54:32.347432270Z",
    "level": "error",
    "source": "atc",
    "message": "atc.pipelines.scheduler.tick.scheduling.try-start-next-pending-build.create.failed-to-fetch-resource",
    "data": {
        "build-id": 43642,
        "build-name": "3",
        "error": "resource script '/opt/resource/in [/tmp/build/get]' failed: exit status 1",
        "pipeline": "<pipeline>",
        "session": "18.332.17527.3.9.140",
        "team": "<team>"
    }
}
```

Might be that the `git` resource is failing also, and that we just don't see the error messages https://github.com/concourse/git-resource/issues/252?

#### Version info

Concourse version: 5.1.0
Deployment type (BOSH/Docker/binary): Binary on EC2.
Infrastructure/IaaS: AWS
Browser (if applicable): N/A
Did this used to work?: Yes, until we upgraded to 5.1.0 from 4.2.3.

**Comment by clarafu** [comment]
@itsdalmo Have you observed that there no new versions of that git resource being fetched which is causing the jobs to not trigger or is it that concourse has fetched those new versions but the jobs are not automatically triggering?

**Comment by itsdalmo** [comment]
@clarafu - for the situation described above, there were new versions not being picked up by the `git` resource for 1/4 resources pointing to different subdirectories of the same repository. I also attempted to manually trigger a job which is supposed to be triggered by the failing `git` resource, and it just got stuck as "pending" for ~4 days before I aborted it just now. 

Also worth noting that for all the resources in this pipeline the "last checked" time is listed as between 3 and 5 days, even though `--resource-checking-interval` is not set on the ATC and `check_every` for the resources are either left at default or set to `1m` in the pipeline definition (checked this with `get-pipeline`). Probably worth noting that we are running Concourse 5.1.0 without global resources, i.e. did _not_ pass `--enable-global-resources` when starting our ATC's.

As I was writing this comment I did another `check-resource` on the stuck resource from my original comment, and over the weekend it seems like the resource has become "unstuck" by itself (the stuck build container mentioned in my last comment seems to have been GC'd as well) - so now the manual check ran successfully and it discovered the versions that were previously missing. However, the dependent job is not being triggered by discovering the new versions - the GUI says that the last job is pending even though I aborted it, but even so I would expect a build to be scheduled and just waiting for `max_in_flight` if it was set, but nothing is happening. The pipeline in question has been deleted and recreated several times, and works great in the start, but then eventually gets stuck. 

Sorry if I'm not describing the problem very well, but there seems to be a lot of strangeness going on with our Concourse deployment since upgrading (specifically around resources/versions). I have some other pipelines that are acting strangely that I can take a look at and see if there is any commonality with the issue describe above, and then post a comment here accordingly. Thanks!
